app/uk/gov/hmrc/agentsubscription/auth/AuthActions.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.agentsubscription.auth
+
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import play.api.mvc._
+import uk.gov.hmrc.agentsubscription.connectors.AuthConnector
+import uk.gov.hmrc.play.http.HeaderCarrier
+
+import scala.concurrent.Future
+
+trait AuthActions {
+  me: Results =>
+
+  val authConnector: AuthConnector
+
+  protected val agentWithEnrolments = new ActionBuilder[RequestWithEnrolments] with ActionRefiner[Request, RequestWithEnrolments] {
```

Comment:
The name `agentWithEnrolments` makes my brain want to think that this is checking that the agent has enrolments. Whereas what it actually does is checks that affinityGroup = Agent and just makes the enrolments available in the augmented request object.

`AuthorisedWithSubscribingAgent` in agent-subscription-frontend does something similar, perhaps that would be a good name here too?